dot/core/service.go

```diff
@@ -350,15 +351,17 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 				continue
 			}
 
-			externalExt := make(types.Extrinsic, 0, 1+len(ext))
-			externalExt = append(externalExt, byte(types.TxnExternal))
-			externalExt = append(externalExt, ext...)
-			txv, err := rt.ValidateTransaction(externalExt)
+			externalExt, err := s.buildExternalTransaction(rt, ext)
+			if err != nil {
+				return fmt.Errorf("building external transaction: %w", err)
```

Comment:
why not `continue`? This is essentially stopping the chain reorg.

Reply:
really good catch